src/codegen/compile_classical.ml

```diff
@@ -12187,6 +12215,29 @@ and compile_prim_invocation (env : E.t) ae p es at =
     SR.Vanilla, Cycles.refunded env
   | SystemCyclesBurnPrim, [e1] ->
     SR.Vanilla, compile_exp_vanilla env ae e1 ^^ Cycles.burn env
+  | ICCyclesPrim, [] ->
```

Comment:
This is not an idempotent operation, so we have to be careful to not call it twice. E.g. it fails for paired up environment+parenthetical.

Reply:
This should return two options. Possibly just the `Nat` (`cycles`) and `Nat32` (`timeout`). We definitely have to search the attributes.